xscen/ensembles.py

```diff
@@ -441,6 +475,83 @@ def generate_weights(
         # Drop the experiment coordinate
         weights = weights.drop_vars("experiment")
 
+    # Attribute_weights
+    if attribute_weights:
+        stationary_weights = {}
+        non_stationary_weights = {}
+        for att, v_att in attribute_weights.items():
+            # Add warning when a mismatch between independance_level/experiment_weight and attribute_weights is detected
+            if att == "experiment" and not balance_experiments:
+                warnings.warn(
+                    "Key experiment given in attribute_weights without argument balance_experiments=True"
+                )
+
+            if (
+                att in ["driving_model", "source"]
+                and independence_level not in ["model", "GCM"]
+            ) or (att == "institution" and independence_level != "institution"):
```

Comment:
I'd be as explicit as possible here.
```suggestion
            if ((att == "source" and independence_level != "model") or (att == "driving_model" and independence_level != "GCM") or (att == "institution" and independence_level != "institution"):
```